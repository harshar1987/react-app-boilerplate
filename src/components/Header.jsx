import React from "react";
import PropTypes from "prop-types";

export const Header = (props) => {
  return  <nav className="header navbar navbar-dark bg-dark">
        <div className="container">
            <div className="row m-auto">
                <i className="fa fa-film fa-2x text-white my-auto"></i>
                <div className="h3 ml-3 my-auto text-light" href="/">{props.title}</div>
            </div>
        </div>
    </nav>;
};

Header.propTypes = {
    title: PropTypes.string
};

