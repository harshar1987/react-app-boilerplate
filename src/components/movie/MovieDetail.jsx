import React from "react";
import { MovieService } from "../../lib/services/MovieService";

function Repeat(props) {

    let items = [];
    const movieDetail = props.movieData;
    for (const key in movieDetail) {
        if (movieDetail.hasOwnProperty(key)) {
            items.push(props.children(key));
        }
    }

    return <div>{items}</div>;
}

export const MovieDetail = (props) => {

    const movieId = props.location.state.movieId;
    const movieDetail = MovieService.getMovieById(movieId);

    return <Repeat movieData={movieDetail} >
        {
            (key) => <div key={key} className="row" style={{padding: "20px"}}>
                        <div className="col-sm-4 clear-fix" style={{ backgroundColor: "lavender", textAlign: "center" }}>{key}</div>
                        <div className="col-sm-4 clear-fix" style={{ backgroundColor: "lavender", textAlign: "center", overflow:"hidden" }}>{movieDetail[key]}</div>
                    </div>
        }
    </Repeat>;
};