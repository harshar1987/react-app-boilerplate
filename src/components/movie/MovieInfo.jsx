import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import MovieInfoStyles from "../../styles/components/movie/movieInfo.css";

export class MovieInfo extends Component {
    constructor(props) {
        super(props);

        this.state = {
            redirect: false,
            movieId: props.id
        };

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        this.setState({ redirect: true });
    }

    renderButton() {
        const btnClass = `btn btn-info ${MovieInfoStyles.movieInfoBtn}`;
        return (
            <div style={{ float: "right", paddingTop: "5px" }}>
                <button type="button" className={btnClass} onClick={this.handleClick}>
                    <i>More Info...</i>
                </button>
            </div>
        );
    }

    render() {
        if (this.state.redirect) {
            return <Redirect push to={{ pathname: "/detail", state: this.state }} />;
        }

        return this.renderButton();
    }
}

