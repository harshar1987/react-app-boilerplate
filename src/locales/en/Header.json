{
    "Movie Cards": "Movie Cards in english"
}