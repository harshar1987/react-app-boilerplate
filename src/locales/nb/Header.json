{
    "Movie Cards": "Filmkort in norwegian"
}