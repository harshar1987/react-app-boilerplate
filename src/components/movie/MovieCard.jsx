import React from "react";
import PropTypes from "prop-types";
import { StarRating } from "./StarRating";
import MovieCardStyles from "../../styles/components/movie/movieCard.css";
import MovieInfo from "./MovieInfo";
import { MovieCardClassNames } from "./styles/MovieCard-Style";
import {WithTranslation} from "si-react-localize";

const MovieCard = (props) => {
    const rating = Number(props.movie.rating);
    const cardDescriptionStyle = {
        fontSize: "14px",
        overflow: "hidden",
        height: "200px",
        textOverflow: "ellipsis"
    };

    return <div className={MovieCardStyles.movieCard}>
        <div className={MovieCardStyles.movieCard} style={{ height: "450px" }}>
            <div style={{ padding: "1.25rem" }}><img className={MovieCardClassNames.cardImgTop} src={props.movie.poster} alt="" style={{ height: "400px" }} /></div>
            <div className={MovieCardClassNames.cardBody}>
                <h4 className={MovieCardClassNames.cardTitle}>{props.movie.title} ({props.movie.year})</h4>
                <h6 className={MovieCardClassNames.cardSubTitle}>
                    {
                        props.movie.genre.split(",").map(cat => props.localize(`Genre.${cat.trim()}`))
                            .reduce((prev, curr) => [prev, ", ", curr])
                    }
                </h6>
                <p className={MovieCardClassNames.cardDescription} style={cardDescriptionStyle}>
                    <b>{props.localize("Plot")}:</b> {props.localize(`Description.${props.movie.title.trim()}`)}
                </p>
            </div>
            <div className={MovieCardClassNames.cardFooter}>
                <div className={MovieCardClassNames.clearFix}>
                    <div className={MovieCardClassNames.starRating}>
                        <StarRating rating={rating} />
                    </div>
                    <div className={MovieCardClassNames.starRatingNumber} style={{ fontSize: "1rem" }}>{rating}</div>
                </div>
                <div className={MovieCardClassNames.clearFix}><MovieInfo id={props.movie.imdb_id} /></div>
            </div>
        </div>
    </div>;
};

MovieCard.defaultProps = {
    movie: {}
};

MovieCard.propTypes = {
    movie: PropTypes.object
};

export default WithTranslation("MovieCard")(MovieCard);