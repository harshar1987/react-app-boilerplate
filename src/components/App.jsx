import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Home from "./Home";
import { MovieDetail } from "./movie/MovieDetail";
import { ErrorBoundary } from "./common/ErrorBoundary";

const App = () =>
    (<ErrorBoundary>
        <Router>
            <Switch>
                <Route exact path='/' component={Home} />
                <Route exact path='/detail' component={MovieDetail} />
            </Switch>
        </Router>
    </ErrorBoundary>);

export default App;



