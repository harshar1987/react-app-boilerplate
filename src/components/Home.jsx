import React from "react";
import { Header } from "./Header";
import { Movies } from "./movie/Movies";
import "../styles/site.css";

export const Home = (props) => {
    
    return (
        <div>
            <Header title={props.headerTitle} />
            <div className="mt-5">
                <Movies />
            </div>
        </div>
    );
};

Home.defaultProps = {
    headerTitle: "Movie Cards"
};