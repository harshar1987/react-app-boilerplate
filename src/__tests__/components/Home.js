import React from "react";
import ReactDOM from "react-dom";
import { Home } from "../../components/Home";
import { Header } from "../../components/Header";
import { Movies } from "../../components/movie/Movies";


describe("Movie Cards Home component", () => {

    it("renders without error", () => {
        const app = document.createElement("div");
        ReactDOM.render(<Home />, app);
    });

    it("should have header and movies as child components", () => {
        const homeComponent = global.shallow(<Home />);
        const header = homeComponent.find(Header);
        const movies = homeComponent.find(Movies);

        expect(header).toHaveLength(1);
        expect(movies).toHaveLength(1);
    });

    it("should have header title as per title set in default props",() => {
        const homeComponent = global.mount(<Home />);
        expect(homeComponent.props().headerTitle).toEqual("Movie Cards");
    });

    it("should have header title as per the title passed in props",() =>{

        let title ="IMDB movie cards";
        const homeComponent = global.mount(<Home headerTitle={title}/>);
        expect(homeComponent.props().headerTitle).toEqual(title);

        title = "movie";
        homeComponent.setProps({headerTitle: title});
        expect(homeComponent.props().headerTitle).toEqual(title);

    });

    it("matches the home component snapshot", () => {
        const tree = global.shallow(<Home />);
        expect(tree).toMatchSnapshot();
    });

    it("matches the home component with the provided title", () => {

        const tree = global.shallow(<Home headerTitle="Custom Title" />);
        expect(tree).toMatchSnapshot();
    });
});

