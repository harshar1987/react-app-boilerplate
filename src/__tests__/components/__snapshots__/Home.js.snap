// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Cards Home component matches the home component snapshot 1`] = `
<div>
  <WrappedComponent
    title="Movie Cards"
  />
  <div
    className="mt-5"
  >
    <Movies />
  </div>
</div>
`;

exports[`Movie Cards Home component matches the home component with the provided title 1`] = `
<div>
  <WrappedComponent
    title="Custom Title"
  />
  <div
    className="mt-5"
  >
    <Movies />
  </div>
</div>
`;
